recipes/fmt/all/conanfile.py

```diff
@@ -71,58 +63,50 @@ def configure(self):
-        if self._cmake:
-            return self._cmake
-        self._cmake = CMake(self)
-        self._cmake.definitions["FMT_DOC"] = False
-        self._cmake.definitions["FMT_TEST"] = False
-        self._cmake.definitions["FMT_INSTALL"] = True
-        self._cmake.definitions["FMT_LIB_DIR"] = "lib"
-        if self._has_with_os_api_option:
-            self._cmake.definitions["FMT_OS"] = self.options.with_os_api
-        self._cmake.configure(build_folder=self._build_subfolder)
-        return self._cmake
+        get(self, **self.conan_data["sources"][str(self.version)], strip_root=True)
 
     def build(self):
-        for patch in self.conan_data.get("patches", {}).get(self.version, []):
-            tools.patch(**patch)
+        apply_conandata_patches(self)
         if not self.options.header_only:
-            cmake = self._configure_cmake()
+            cmake = CMake(self)
+            # FIXME : https://github.com/conan-io/conan/issues/11476
+            # can be replaced by https://docs.conan.io/en/latest/reference/conanfile/tools/cmake/cmaketoolchain.html#cache-variables in 1.50
+            cache_entries = {
+                "FMT_DOC": "False",
+                "FMT_TEST": "False",
+                "FMT_INSTALL": "True",
+                "FMT_LIB_DIR": "lib"
+            }
+            if self._has_with_os_api_option:
+                cache_entries["FMT_OS"] = self.options.with_os_api
+            cmake.configure(variables=cache_entries)
             cmake.build()
 
     def package(self):
-        self.copy("LICENSE.rst", dst="licenses", src=self._source_subfolder)
+        copy(self, pattern="*LICENSE.rst", src=self.source_folder, dst=os.path.join(self.package_folder, "licenses"))
         if self.options.header_only:
-            self.copy("*.h", dst="include", src=os.path.join(self._source_subfolder, "include"))
+            copy(self, pattern="*.h", src=os.path.join(self.source_folder, "include"), dst=os.path.join(self.package_folder, "include"))
         else:
-            cmake = self._configure_cmake()
+            cmake = CMake(self)
             cmake.install()
-            tools.rmdir(os.path.join(self.package_folder, "lib", "cmake"))
-            tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
-            tools.rmdir(os.path.join(self.package_folder, "share"))
+            rmdir(self, os.path.join(self.package_folder, "lib", "cmake"))
+            rmdir(self, os.path.join(self.package_folder, "lib", "pkgconfig"))
+            rmdir(self, os.path.join(self.package_folder, "res"))
+            rmdir(self, os.path.join(self.package_folder, "share"))
 
     def package_info(self):
         self.cpp_info.names["cmake_find_package"] = "fmt"
         self.cpp_info.names["cmake_find_package_multi"] = "fmt"
         self.cpp_info.names["pkg_config"] = "fmt"
         if self.options.header_only:
+            self.cpp_info.components["fmt-header-only"].includedirs.extend(["include"])
```

Comment:
I don't understand this creation of a component. Why when the recipe is header-only is necessary to create a component? Components with a header-only library doesn't make sense, and having one recipe with just 1 component doesn't have much sense either?

Reply:
I think this is related to the target name in cmake?
In that case, maybe we should change that to create a component with _ in the name and then using the .names on the component to seet that to fmt::fmt-header-only ? And also use the set_property.
And also a comment would be helpful because this an be confusing for users.

Reply:
maybe it was necessary in the past to get some desired behavior with `FindFMT.cmake` or `fmt-config.cmake` to match upstream, e.g. to bypass some `cmake_find_package[_multi]` limitation.
again, it's not something new, but maybe @czoido remembers.